src/CMakeLists.txt

```
@@ -1,171 +1,172 @@
-    widgets/subtoolwidget.cpp
-    widgets/keybuttonwidget.cpp
-    widgets/sidebar.cpp
-    widgets/shottoolwidget.cpp
-    widgets/colortoolwidget.cpp
-    dbusinterface/dbusnotify.cpp
-    dbusinterface/dbuszone.cpp
-    dbusinterface/dbuscontrolcenter.cpp
-    dbusinterface/dbussoundeffect.cpp
-    dbusservice/dbusscreenshotservice.cpp
-    dbusservice/dbusscreenshot.cpp
-    widgets/camerawidget.cpp
-    utils/dbusutils.cpp
-    screenshot.cpp
-    utils/voicevolumewatcher.cpp
-    utils/camerawatcher.cpp
-    widgets/tooltips.cpp
-    widgets/filter.cpp
-    utils/desktopinfo.cpp
-    utils/screengrabber.cpp
-    dbusinterface/drawinterface.cpp
-    screen_shot_event.cpp
-    RecorderRegionShow.cpp
-    xgifrecord.cpp
-    recordertablet.cpp
-    )
+find_package(DFrameworkdbus REQUIRED)
+find_package(PkgConfig REQUIRED)
+#find_package(Qt5Widgets REQUIRED)
+find_package(DtkWidget REQUIRED)
+find_package(DtkCore REQUIRED)
+find_package(DtkGui REQUIRED)
+find_package(DdeDock REQUIRED)
+#find_package(DtkCMake REQUIRED)
+pkg_check_modules(gstreamer-audio REQUIRED gstreamer-audio-1.0)
+pkg_check_modules(libffmpegthumbnailer REQUIRED libffmpegthumbnailer)
+pkg_check_modules(libusb REQUIRED libusb-1.0)
+pkg_check_modules(X11 REQUIRED x11)
+pkg_check_modules(XEXT REQUIRED xext)
+pkg_check_modules(XFIXES REQUIRED xfixes)
+pkg_check_modules(libimagevisualresult REQUIRED libimagevisualresult)
+pkg_check_modules(Xtst REQUIRED xtst)
+add_definitions(-DAPP_VERSION="${PROJECT_VERSION}")
 
-set ( deepin-screen-recorder_RSCS
-    ../assets/image/deepin-screen-recorder.qrc
-    ../assets/resources/resources.qrc
-    ../assets/icons/icons.qrc
-    )
+set(deepin-screen-recorder_HDRS
+  ${glob_HEAD}
+  ${accessibility_HEAD}
+  ${menucontroller_HEAD}
+  ${camera_HEAD}
+  ${dbusinterface_HEAD}
+  ${dbusserver_HEAD}
+  ${utils_HEAD}
+  ${widgets_HEAD}
+  ${gstrecord_HEAD}
+)
 
-add_executable(${PROJECT_NAME}
-    ${deepin-screen-recorder_HDRS} ${deepin-screen-recorder_SRCS} ${deepin-screen-recorder_RSCS}
-    )
+set(deepin-screen-recorder_SRCS
+  main.cpp
+  ${glob_SRC}
+  ${menucontroller_SRC}
+  ${camera_SRC}
+  ${dbusinterface_SRC}
+  ${dbusserver_SRC}
+  ${utils_SRC}
+  ${widgets_SRC}
+  ${gstrecord_SRC}
+)
 
-qt5_use_modules(${PROJECT_NAME} ${QtModule})
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(3rd_lib REQUIRED
-    dtkwidget dtkgui dframeworkdbus xcb xcb-util
-    )
-target_include_directories(${PROJECT_NAME} PUBLIC ${3rd_lib_INCLUDE_DIRS} )
-target_link_libraries(${PROJECT_NAME} ${3rd_lib_LIBRARIES} ${LINK_DIR} )
+set (deepin-screen-recorder_RSCS
+  ../assets/image/deepin-screen-recorder.qrc
+  ../assets/resources/resources.qrc
+  ../assets/icons/icons.qrc
+)
+set(COMPILE_SRC
+  ${libcam_TOP_SRC}
+  ${deepin-screen-recorder_HDRS}
+  ${deepin-screen-recorder_SRCS}
+  ${libcam_TOP_HEAD}
+  ${deepin-screen-recorder_RSCS}
+  ${TRANSLATEDFILES}
+)
+set(include_DIRS 
+  #../../interfaces/
+  ${DFrameworkdbus_INCLUDE_DIRS}
```

```suggestion
  ${DFrameworkdbus_INCLUDE_DIR}
```
https://github.com/linuxdeepin/dde-qt-dbus-factory/blob/master/libdframeworkdbus/DFrameworkdbusConfig.in

带 s 的是用的 pkg-config